lib/fathead/php/parse.js

```diff
@@ -0,0 +1,189 @@
+    const jsdom = require('jsdom');
+
+    const DOWNLOAD_DIR = 'download';
+    const DOC_DIR = DOWNLOAD_DIR + '/phpdoc';
+
+    const BASE_URL = 'http://php.net/manual/en/';
+
+    const OUTPUT_FILE = 'output.txt'
+
+    // remove output file before parse
+    try {fs.unlinkSync(OUTPUT_FILE)} catch(e) {};
+
+    function addItem(item) {
+
+        const defaultItem = {
+            name: '',
+            type: 'A',
+            redirect: '',
+            ignore1: '',
+            categories: '',
+            ignore2: '',
+            related: '',
+            ignore3: '',
+            external_links: '',
+            disambiguation: '',
+            image: '',
+            abstract: '',
+            url: ''
+        }
+
+        for (let key in defaultItem) {
+            item[key] = item[key] || defaultItem[key]
+        }
+
+        const line = [
+            item.name,
+            item.type,
+            item.redirect,
+            item.ignore1,
+            item.categories,
+            item.ignore2,
+            item.related,
+            item.ignore3,
+            item.external_links,
+            item.disambiguation,
+            item.image,
+            item.abstract,
+            item.url,
+            "\n"
+        ].join("\t")
+
+        fs.appendFile(OUTPUT_FILE, line, (err) => {
+            if (err) console.error('OUTPUT:', err);
+        });
+    }
+
+    // function which read all files from phpdoc and call it parser
+    function main() {
+
+        function fileIterator(file) {
+
+            if (ignoredFiles.indexOf(file) > -1) return;
+
+            const pattern = /(^.*)\.(.*)\.html$/;
+            const matches = file.match(pattern);
+            const type = matches[1] || '';
+            const name = matches[2] || '';
+
+            // parse documentation
+            if (parsers[type]) {
+                parsers[type](name, file);
+            }
+        }
+
+        fs.readdir(DOC_DIR, (err, files) => {
+
+            if (err) {
+                console.error('DIRECTORY LIST: ', err);
+                return;
+            }
+
+            files.forEach(fileIterator);
+        });
+    }
+
+    // function to parse all php functions
+    function parserFunction(key, fileName) {
+
+        jsdom.env(DOC_DIR + '/' + fileName, (err, window) => {
+
+            if (err) {
+                console.error(err)
+                return;
+            }
+
+            const none = {textContent : ''}
+
+            const name = window.document.querySelector('.refname') || none;
+            const version = window.document.querySelector('.verinfo') || none;
+            const title = window.document.querySelector('.dc-title') || none;
+            const description = window.document.querySelector('.description .dc-description') || none;
+
+            let abstract = '';
+            abstract += version.textContent + ' — ' + title.textContent;
+            abstract += '<pre><code>' + description.textContent + '</code></pre>';
```

Comment:
It looks like we need to trim `description.textContent` -- it seems to always have 4 leading spaces